pkg/kubelet/kubelet.go

```diff
@@ -2138,6 +2138,9 @@ func (kl *Kubelet) cleanUpContainersInPod(podId types.UID, exitedContainerID str
 			removeAll = eviction.PodIsEvicted(syncedPod.Status)
 		}
 		kl.containerDeletor.deleteContainersInPod(exitedContainerID, podStatus, removeAll)
+		if kl.gpuManager != nil {
+			kl.gpuManager.UpdateDevices(string(podId), exitedContainerID)
```

Comment:
Make UpdateDevices take a types.UID for the podID rather than casting to string here